### ISSUE TITLE
Roadmap / Comparison to other Go ML libraries

### ISSUE THREAD

**Issue by derekperkins** [cdipaolo/goml#3]
How does goml compare to some of the other Go libraries in terms of product vision / roadmap?
 - https://github.com/sjwhitworth/golearn
 - https://github.com/alonsovidales/go_ml

There's a decent amount of overlap in terms of the implemented algorithms / models. Is your goal to eventually include all of the other types (neural networks, collaborative filtering, etc)? It seems like the stated goal of being more stream oriented than batch oriented differentiates this library too.

At the end of the day, this seems like the most active repo with an exciting direction. I'm very curious to know where you plan on taking things.

**Comment by cdipaolo** [comment]
Yeah I'd like to implement a bunch of other types of models. I'll write up
a roadmap when I have some free time (currently adding k-means++
instantiation to the batch k-means.)

I hope to base this library on online models/versions of models, but I want
to give the optionality to use batch methods so people won't have to switch
libraries to work with a potentially better model.

Let me know any questions, opinions, comments, etc. you may have.
